app/commands/v2/patch_link_set.rb

```diff
@@ -38,6 +32,14 @@ def call
 
     private
 
+      def find_or_create_link_set(content_id)
+        begin
+          LinkSet.find_or_create_by!(content_id: content_id).lock!
+        rescue ActiveRecord::RecordNotUnique
+          retry
```

Comment:
It seems a bit dangerous to always retry here; a deadlock might mean this continues indefinitely. Instead it would be better to keep a count and retry a maximum of, say, 3 times.